### PR TITLE
GODRIVER-2674 use time.Duration instead of durationNanos

### PR DESCRIPTION
**Pull request by jtagcat**
Same as #1105

```
OK make fmt
OK make lint
make test: go.mongodb.org/mongo-driver/mongo/integration/unified: no such command: 'configureFailPoint'
(make test fails on master with same result)
make test-race: same result as make test
```

### CODE REVIEW

**Review by qingyang-hu — COMMENTED** [review]
The replacement seems reasonable though it is backward-breaking.
Instead, in v1.x, we can deprecate the current fields while adding new fields named `Duration` that are type `time.Duration`.
We can retire these `DurationNanos` fields in the upcoming v2 driver.

File: mongo/integration/unified/client_entity.go [hunk 1]
```diff
@@ -291,7 +293,7 @@ func (c *clientEntity) processFailedEvent(_ context.Context, evt *event.CommandF
 	bsonBuilder := bsoncore.NewDocumentBuilder().
 		AppendString("name", "CommandFailedEvent").
 		AppendDouble("observedAt", getSecondsSinceEpoch()).
-		AppendInt64("durationNanos", evt.DurationNanos).
+		AppendInt64("duration", int64(evt.Duration)).
```
Comment (jtagcat) [inline mongo/integration/unified/client_entity.go:296]: How is the bson used? Why do other events not use Duration (non-exhaustive)?

**Review by matthewdale — CHANGES_REQUESTED** [review]
Hey @jtagcat sorry about the slow review and thanks for the edits! I have one required change, but it otherwise looks great!

**Comment by matthewdale** [comment]
Closing in favor of https://github.com/mongodb/mongo-go-driver/pull/1193

**Pull request closed without merging**